src/backend/storeManagers/legendary/setup.ts

```diff
@@ -30,107 +27,27 @@ export const legendarySetup = async (appName: string) => {
     protonVerb: 'waitforexitandrun'
   })
 
-  // if not a ubisoft game, do nothing
-  if (gameInfo.install.executable !== 'UplayLaunch.exe') {
-    return
-  }
-
-  if (await isUbisoftInstalled(appName)) {
-    // it's already installed, do nothing
-    return
-  }
-
-  sendGameStatusUpdate({
-    appName,
-    runner: 'legendary',
-    status: 'ubisoft'
-  })
-
-  await installUbisoftConnect(appName)
-}
-
-const installUbisoftConnect = async (appName: string) => {
-  try {
-    await downloadIfNotCached(cachedUbisoftInstallerPath, UBISOFT_INSTALLER_URL)
-
-    await runWineCommandOnGame(appName, {
-      commandParts: [cachedUbisoftInstallerPath, '/S']
-    })
-  } catch (error) {
-    logWarning(`Error installing Ubisoft Connect: ${error}`, LogPrefix.Backend)
-  }
-
-  if (await isUbisoftInstalled(appName)) {
-    return true
-  } else {
-    // it was not installed correctly, show an error
-    showDialogBoxModalAuto({
-      title: i18next.t('box.error.ubisoft-connect.title', 'Ubisoft Connect'),
-      message: i18next.t(
-        'box.error.ubisoft-connect.message',
-        'Installation of Ubisoft Connect in the game prefix failed. Check our wiki page at https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/wiki/How-to-install-Ubisoft-Connect-on-Linux-and-Mac to install it maunally.'
-      ),
-      type: 'ERROR'
-    })
-    return false
-  }
-}
-
-const downloadIfNotCached = async (cachePath: string, url: string) => {
-  if (existsSync(cachePath)) {
-    // if the cached file exist but it was cached more than a week ago, download a new one
-    // an outdated installer can make the installation fail according to some report
-    const cachedAt = statSync(cachePath).mtime
-    const oneWeekAgo = new Date()
-    oneWeekAgo.setDate(oneWeekAgo.getDate() - 7)
-    if (cachedAt > oneWeekAgo) {
-      return true
+  if (
+    gameInfo.install?.platform === 'Windows' ||
+    gameInfo.install?.platform === 'Win32' ||
+    gameInfo.install?.platform === 'windows'
+  ) {
+    const info = await getInstallInfo(appName, gameInfo.install.platform)
```

Comment:
this call to getInstallInfo can throw an exception, in other places in the code we do a try/catch just in case

Reply:
where? I don't see a try catch anywhere

Reply:
there's a try catch within the function itself it looks like

Reply:
The function has a try/catch inside, but when it catches an error it throws a new one https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/a460b9721910a7a55ed59ccb15aae9992dad15ec/src/backend/storeManagers/legendary/library.ts#L245

I found one getInstallInfo using the try/catch here https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/a460b9721910a7a55ed59ccb15aae9992dad15ec/src/frontend/screens/Library/components/InstallModal/DownloadDialog/index.tsx#L267 but now I see it's a different `getInstallInfo` function, not that one


I do think we should handle that possible throw though (I think the other places that are not handling it are something that we should fix eventually too)

Reply:
makes sense